scalr/resource_scalr_agent_pool_token_test.go

```diff
@@ -0,0 +1,240 @@
+				Config:    testAccScalrAgentPoolTokenChangedOutside(pool),
+				PlanOnly:  true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("scalr_agent_pool_token.test", "description", "changed-outside-of-terraform"),
+					resource.TestCheckResourceAttr("scalr_agent_pool_token.test", "agent_pool_id", pool.ID),
+				),
+			},
+		},
+	})
+}
+func TestAccScalrAgentPoolToken_update(t *testing.T) {
+	var pool scalr.AgentPool
+	if isAccTest() {
+		pool = createPool(t)
+		defer deletePool(t, pool)
+	}
+	token := &scalr.AgentPoolToken{}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckScalrAgentPoolTokenDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccScalrAgentPoolTokenBasic(pool),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("scalr_agent_pool_token.test", "description", "agent_pool_token-test"),
+					resource.TestCheckResourceAttr("scalr_agent_pool_token.test", "agent_pool_id", pool.ID),
+				),
+			},
+
+			{
+				Config: testAccScalrAgentPoolTokenUpdate(pool),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScalrAgentPoolTokenExists("scalr_agent_pool_token.test", pool, token),
+					resource.TestCheckResourceAttr("scalr_agent_pool_token.test", "description", "agent_pool_token-updated"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckScalrAgentPoolTokenExists(resId string, pool scalr.AgentPool, token *scalr.AgentPoolToken) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		scalrClient := testAccProvider.Meta().(*scalr.Client)
+
+		rs, ok := s.RootModule().Resources[resId]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resId)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		// Get the token
+		l, err := scalrClient.AgentPoolTokens.List(ctx, pool.ID, scalr.AgentPoolTokenListOptions{})
+		if err != nil {
+			return err
+		}
+
+		if len(l.Items) != 1 {
+			return fmt.Errorf("There are more than one token for pool: %d", len(l.Items))
+		}
+		if l.Items[0].ID != rs.Primary.ID {
+			return fmt.Errorf("Expected to get: %s, got: %s.", rs.Primary.ID, l.Items[0].ID)
+		}
+
+		*token = *l.Items[0]
+
+		return nil
+	}
+}
+
+func testAccCheckScalrAgentPoolTokenChangedOutside(pool scalr.AgentPool, token *scalr.AgentPoolToken) func() {
+	return func() {
+		scalrClient := testAccProvider.Meta().(*scalr.Client)
+
+		r, err := scalrClient.AccessTokens.Update(
+			context.Background(),
+			token.ID,
+			scalr.AccessTokenUpdateOptions{Description: scalr.String("changed-outside-of-terraform")},
+		)
+		if err != nil {
+			log.Fatalf("Could not update the agent pool outside of terraform: %v", err)
+		}
+
+		if r.Description != "changed-outside-of-terraform" {
+			log.Fatalf("Failed to update the agent pool outside of terraform: %v", err)
+		}
+	}
+}
+
+func createScalrClient() (*scalr.Client, error) {
```

Comment:
Should be moved to `testing.go` as it's not specific to agents. There is a client create function there already, but it's for mock clients.